cmd/ci-reporter/cmd/testgrid.go

```diff
@@ -17,97 +17,84 @@ limitations under the License.
+			testgrid.DashboardName(fmt.Sprintf("sig-release-%s-blocking", cfg.ReleaseVersion)),
+			testgrid.DashboardName(fmt.Sprintf("sig-release-%s-informing", cfg.ReleaseVersion)),
 		}...)
 	}
-	result := make([]TestgridStatistics, 0)
-	for _, job := range requiredJobs {
-		resp, err := http.Get(fmt.Sprintf("https://testgrid.k8s.io/%s/summary", job.URLName))
-		if err != nil {
-			return nil, err
-		}
-		defer resp.Body.Close()
 
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return nil, err
-		}
+	return testgrid.ReqTestgridDashboardSummaries(testgridURLs)
+}
 
-		jobs := make(map[string]overview)
-		err = json.Unmarshal(body, &jobs)
+// PrintTestgridReportData used to print testgrid report data out to the console
+func PrintTestgridReportData(cfg ReporterConfig, stats *testgrid.DashboardData) error {
+	printShortReport := func(name testgrid.DashboardName, jobData *testgrid.JobData) error {
+		overview, err := jobData.Overview()
 		if err != nil {
-			return nil, err
+			return errors.Wrap(err, "could not get testgrid data overview")
 		}
-
-		statistics := getStatistics(jobs)
-		statistics.Name = job.OutputName
-		result = append(result, statistics)
+		fmt.Printf("\nOverview for %s\n", name)
+		fmt.Printf("\t%d jobs total\n", len(overview.FailingJobs)+len(overview.FlakyJobs)+len(overview.PassingJobs)+len(overview.StaleJobs))
+		fmt.Printf("\t%d are passing\n", len(overview.PassingJobs))
+		fmt.Printf("\t%d are flaking\n", len(overview.FlakyJobs))
+		fmt.Printf("\t%d are failing\n", len(overview.FailingJobs))
+		if len(overview.StaleJobs) > 0 {
+			fmt.Printf("\t%d are stale\n", len(overview.StaleJobs))
+		}
+		return nil
 	}
-	return result, nil
-}
 
-// PrintTestgridReportData used to print testgrid report data out to the console
-func PrintTestgridReportData(stats []TestgridStatistics) {
-	for _, stat := range stats {
-		fmt.Printf("Failures in %s\n", stat.Name)
-		fmt.Printf("\t%d jobs total\n", stat.Total)
-		fmt.Printf("\t%d are passing\n", stat.Passing)
-		fmt.Printf("\t%d are flaking\n", stat.Flaking)
-		fmt.Printf("\t%d are failing\n", stat.Failing)
-		fmt.Printf("\t%d are stale\n", stat.Stale)
-		fmt.Print("\n\n")
+	printLongReport := func(name testgrid.DashboardName, jobData *testgrid.JobData) {
+		fmt.Printf("\nDetails for %s\n", name)
+		for jobName := range *jobData {
+			j := *jobData
+			jobData := j[jobName]
+			if jobData.OverallStatus != testgrid.Passing {
+				fmt.Printf("%s: %s\n", jobData.OverallStatus, jobName)
+				fmt.Printf("- %s\n", jobData.GetJobURL(jobName))
+				fmt.Printf("- %s\n", jobData.Status)
+			}
+			if jobData.OverallStatus == testgrid.Failing {
+				// Filter sigs
+				sigRegex := regexp.MustCompile(`sig-[a-zA-Z]+`)
+				sigsInvolved := map[string]int{}
+				for i := range jobData.Tests {
+					sigs := sigRegex.FindAllString(jobData.Tests[i].TestName, -1)
+					for _, sig := range sigs {
+						sigsInvolved[sig]++
+					}
+				}
+				sigs := reflect.ValueOf(sigsInvolved).MapKeys()
```

Comment:
Let's iterate this one and save the whole reflect import:
```suggestion
				sigs := []string{}
				for k := range sigsInvolved {
					sigs = append(sigs, k)
				}
```